### ISSUE TITLE
[FEATURE] I want you correctly corresponding to the XmlSerializer

### ISSUE THREAD

**Issue by satochibi** [mono/SkiaSharp#988]
SkiaSharp can output images and output to svg.
However, what I want to do is to save the picture drawn with SkiaSharp as a proprietary file.
To achieve this, I want to save the object state as XML with C # XmlSerializer.
However, I noticed a problem that XmlSerializer does not support some SkiaSharp classes and structures (Ex. SKColor, SKBitmap, SKPath...).
I would be happy if you responded quickly.
Thank you.

**Comment by mattleibow** [comment]
Thanks for taking the time to open this issue. This may not be exactly in the scope of SkiaSharp. We currently support exporting PNG, JPEG and WEBP files. There is also PDF if that is desired. In addition to those, there is experimental SVG and XPS file formats.

Currently, the SkiaSharp objects are not serializable since this is more a binding than an implementation. What this means is that for every object you see, there may be one or more native object that live in the C++ world. Unfortunately, this makes it hard to support serialization.

What I would be more interested in, is why exporting to PNG or one of the other formats aren't working?

**Comment by BretJohnson** [comment]
@satochibi Can you say more about what your app does & how you plan to use this feature. My https://github.com/BretJohnson/XGraphics may be a better match for what you want to do, as those graphics primitives are serializable (at least that's the intent). It's a higher level abstraction. But mostly, I'm just curious about your use case. Thanks.

**Comment by satochibi** [comment]
Thank you for answering.
I asked the question that I wanted a mechanism to save the information of SkiaSharp objects that are currently painting.
In order to save it, I want it to be realized by some kind of serialization, but can I do it somehow other than XmlSerializer?

**Comment by satochibi** [comment]
> What I would be more interested in, is why exporting to PNG or one of the other formats aren't working?

This is because we want to create drawing software that can handle screen tones of 1px physical units.
At this time, I want to be able to display the screen tone in a certain size, whether the picture drawn on the art board is enlarged or reduced.
We did this, but we considered XML serialization to save and load the state of the picture we drew.
In short, I wanted to do a “Save Project” like an adobe illustrator ai file.

> Unfortunately, this makes it hard to support serialization.

XmlSerializer was difficult, but is binary or json difficult too?

Implemented application:
https://github.com/satochibi/UgomemoPro
